datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -289,13 +289,13 @@ def create_scraper_configuration(self, instance=None):
 
         return config
 
-    def get_http_handler(self, scraper_config):
+    def get_http_handler(self, scraper_config, recreate=False):
         """
         Get http handler for a specific scrapper config.
         The http handler is cached using `prometheus_url` as key.
         """
         prometheus_url = scraper_config['prometheus_url']
-        if prometheus_url in self._http_handlers:
+        if not recreate and prometheus_url in self._http_handlers:
```

Comment:
Could you add a test to test this feature? (if we don't do the other way suggested here https://github.com/DataDog/integrations-core/pull/5455/files#r365948418)